torchdynamo/debug_utils.py

```diff
@@ -159,32 +176,70 @@ def inductor_fails(fx_g, args, check_str=None):
     return False
 
 
-def dump_to_minify(gm, args):
+def nvfuser_fails(fx_g, args, check_str=None):
+    from torch.fx.passes.backends.nvfuser import NvFuserBackend
+
+    nvfuser = NvFuserBackend()
+
+    try:
+        compile_mod = nvfuser(fx_g, args)
+        compile_mod = compile_mod(*args)
+    except Exception as e:
+        if check_str is not None and check_str not in repr(e):
+            return False
+        print(repr(e))
+        return True
+    return False
+
+
+def dump_to_minify(gm, args, compiler_name: str):
     with open(
-        os.path.join(torchdynamo.config.base_dir, "minimizer_repro.py"), "w"
+        os.path.join(torchdynamo.config.base_dir, "minifier_launcher.py"), "w"
     ) as fd:
         fd.write(generate_repro_string(gm, args))
         fd.write("\n")
         fd.write(
             textwrap.dedent(
-                """
+                f"""
                 from functools import partial
-                from torchinductor.debug_utils import (
-                    inductor_fails,
-                    isolate_inductor_fails,
-                    dump_state_inductor,
+                from torchdynamo.debug_utils import (
+                    isolate_fails,
+                    dump_state,
                 )
                 from functorch.compile import minifier
 
-                env_variables = {"CUDA_VISIBLE_DEVICES": "1"}
+                env_variables = {{"CUDA_VISIBLE_DEVICES": "1"}}
```

Comment:
hm, why 1 here?

Reply:
It's useful to run it on a different device to reduce OOM issues (otherwise you're running both the minifier + the checker in the same GPU). In general OOM issues are one thing that we could improve the minifier on.